app/controllers/my_module_repositories_controller.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class MyModuleRepositoriesController < ApplicationController
+  before_action :load_my_module, only: :dropdown_list
+  before_action :check_my_module_view_permissions, only: :dropdown_list
+
+  def dropdown_list
+    @repositories = Repository.accessible_by_teams(current_team)
```

Comment:
You forgot on `Repository.used_on_task_but_unshared`. You can check current implementation of inventories dropdown in `navbar-secondary`

Reply:
I will be in separate task